format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorUtils.java

```diff
@@ -122,30 +125,81 @@ public static String[] setColumnNames(String rawLine, boolean skipHeader,
-   * [A-Za-z] and subsequently contain only [A-Za-z0-9].
+   * [A-Za-z_] and subsequently contain only [A-Za-z0-9_].
    *
-   * @param fieldNames an ar field names to be validated
+   * @param fieldNames an array of field names to be validated
    */
   public static void validateSchemaFieldNames(String[] fieldNames) {
-    String avroNamingStandard = "[A-Za-z_]+[A-Za-z0-9_]*";
-    Pattern pattern = Pattern.compile(avroNamingStandard);
     for (String fieldName : fieldNames) {
-      Matcher matcher = pattern.matcher(fieldName);
+      Matcher matcher = AVRO_NAMING_STANDARD.matcher(fieldName);
       if (!matcher.matches()) {
         throw new IllegalArgumentException(
           String.format(
             "Invalid column name detected: \"%s\"! " +
-            "Column names can only start with capital letters, small letters, and " + "\"_\". " +
-            "Subsequently they may contain only capital letters, small letters, numbers  and \"_\".", fieldName
+              "Column names can only start with capital letters, small letters, and " + "\"_\". " +
+              "Subsequently they may contain only capital letters, small letters, numbers  and \"_\".", fieldName
           )
         );
       }
     }
   }
+
+  /**
+   * Cleans an array of field names to make sure they comply with avro field naming standard.
+   * It also makes sure each name is unique in the list.
+   * Field names can start with [A-Za-z_] and subsequently contain only [A-Za-z0-9_].
+   *
+   * Steps:
+   * 1) Trim surrounding spaces
+   * 2) If its empty replace it with BLANK
+   * 3) If it starts with a number, prepend "col_"
+   * 4) Replace invalid characters with "_" (multiple invalid characters gets replaced with one symbol)
+   * 5) Check if the name has been found before (without considering case)
+   *    if so add _# where # is the number of times seen before + 1
+   * @param fieldNames an array of field names to be cleaned
+   */
+  private static String[] cleanSchemaFieldNames(String[] fieldNames) {
+    final String replacementChar = "_";
+    final List<String> cleanFieldNames = new ArrayList<>();
+    final Map<String, Integer> seenFieldNames = new HashMap<>();
+    for (String fieldName : fieldNames) {
+      StringBuilder cleanFieldNameBuilder = new StringBuilder();
+
+      // Remove any spaces at the end of the strings
+      fieldName = fieldName.trim();
+
+      // If it's an empty string replace it with BLANK
+      if (fieldName.length() == 0) {
+        cleanFieldNameBuilder.append("BLANK");
+      } else if ((fieldName.charAt(0) >= '0') && (fieldName.charAt(0) <= '9')) {
+        // Prepend a col_ if the first character is a number
+        cleanFieldNameBuilder.append("col_");
+      }
+
+      // Replace all invalid characters with the replacement char
+      cleanFieldNameBuilder.append(NOT_VALID_PATTERN.matcher(fieldName).replaceAll(replacementChar));
```

Comment:
Totally nit, but given we have StringBuilder consider java.util.regex.Matcher#appendReplacement